apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -436,13 +489,9 @@ func (m *ModelManagerAPI) CreateModel(ctx context.Context, args params.ModelCrea
 		return result, errors.Annotate(err, "failed to get config")
 	}
 
-	modelType := state.ModelTypeIAAS
-	if jujucloud.CloudIsCAAS(*cloud) {
-		modelType = state.ModelTypeCAAS
-	}
-
-	model, st, err := m.state.NewModel(state.ModelArgs{
-		Type:            modelType,
+	// TODO: remove model creation from the mongo state.
+	_, st, err := m.state.NewModel(state.ModelArgs{
```

Comment:
For the moment I think this is good not to touch. I would suspect that we would follow up pretty quickly with a PR to change ModelArgs so we are proving that we can drop the input down to the bare minimum. i.e getting rid of model config.